ouroboros-consensus/src/Ouroboros/Consensus/HardFork/Combinator/Serialisation/SerialiseNodeToClient.hs

```diff
@@ -45,9 +52,128 @@ import           Ouroboros.Consensus.HardFork.Combinator.Ledger.Query
 import           Ouroboros.Consensus.HardFork.Combinator.Mempool
 import           Ouroboros.Consensus.HardFork.Combinator.Serialisation.Common
 import           Ouroboros.Consensus.HardFork.Combinator.Serialisation.SerialiseDisk ()
+import qualified Ouroboros.Consensus.HardFork.History as History
+
+{-------------------------------------------------------------------------------
+  Serialisation of products
+-------------------------------------------------------------------------------}
+
+-- | Encoding of `NP f xs` while filtering out the components for disabled eras
+-- as indicated by the `HardForkNodeToClientVersion xs`. disabled eras imply
+-- that the protocol version does not support those eras. Hence, omitting the
+-- corresponding elements is the correct behavior.
+encodeNodeToClientNP ::
+     forall xs f. (SerialiseHFC xs)
+  => (   forall x. SerialiseConstraintsHFC x
+      => CodecConfig x
+      -> BlockNodeToClientVersion x
+      -> f x
+      -> Encoding
+     )
+  -- ^ The encoding of the individual elements (assuming era `x` is enabled).
+  -> CodecConfig (HardForkBlock xs)
+  -> HardForkNodeToClientVersion xs
+  -> NP f xs
+  -> Encoding
+encodeNodeToClientNP
+  encodeElement
+  (HardForkCodecConfig (PerEraCodecConfig ccfgs))
+  version
+  xs
+  = case version of
+    HardForkNodeToClientDisabled versionX -> case ccfgs of
+      (ccfg :: CodecConfig x) :* _ -> case xs of
```

Comment:
Is this pattern signature necessary? I generally try to avoid those; they often do counter-intuitive binding. Please remove it, if you can.

Reply:
Will remove